### PR TITLE
netdev: add netdev_trigger_event_isr() function

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR adds a `netdev_irq_end` function that wraps the `dev->event_callback(dev, NETDEV_EVENT_ISR)`call.

This is a step required in order to separate ISR signals from driver events (RX_DONE, TX_DONE), etc. This is useful when network  stacks have different implementations of driver events but they use the same "recv thread".

I will introduce a `netdev_event_thread` module that redefines this function to use the `event_thread` module for handling device ISRs.

### Testing procedure
This should be low hanging fruit. Check if `netdev` drivers still compile (at86rf2xx, ethos).
Running `gnrc_networking` on native should be enough to test functionality.

### Issues/PRs references
None so far

### CODE REVIEW

File: drivers/include/net/netdev.h [hunk 1]
```diff
@@ -469,7 +469,19 @@ static inline int netdev_set_notsup(netdev_t *dev, netopt_t opt,
     return -ENOTSUP;
 }
 
-
+/**
+ * @brief Informs netdev there was an interrupt request from the network device.
+ *
+ *        This function calls @ref event_callback with NETDEV_EVENT_ISR event.
```
Comment (bergzand) [inline drivers/include/net/netdev.h]: The `@ref event_callback` is undefined here (see travis)

Comment (miri64) [reply drivers/include/net/netdev.h]:
```suggestion
 *        This function calls netdev_t::event_callback with NETDEV_EVENT_ISR event.
```

**Comment by bergzand** [comment]
There are a number of remaining calls in:
 - `cpu/esp32/esp-eth/esp_eth_netdev.c`
 - `cpu/esp32/esp-wifi/esp_wifi_netdev.c`
 - `cpu/esp8266/esp-wifi/esp_wifi_netdev.c`
 - `cpu/nrf52/radio/nrf802154/nrf802154.c`
 - `cpu/nrf5x_common/radio/nrfmin/nrfmin.c`

Is there a reason why these are omitted from the cleanup here? Or are they more complex and should they be handled in a follow-up?

**Comment by bergzand** [comment]
And please rebase :)

**Comment by jia200x** [comment]
> There are a number of remaining calls in:

Oh, good catch! I don't know how I missed those. Thanks, I will add them

**Comment by jia200x** [comment]
done!

**Comment by bergzand** [comment]
Looks good, please squash!

**Comment by jia200x** [comment]
squashed :)

**Comment by kaspar030** [comment]
Why `end`?

**Comment by jia200x** [comment]
> Why `end`?

Hmmmm I named it like that because it should be called at the end of IRQ. But strictly speaking it could be called from anywhere during ISR.
Perhaps another name? `netdev_event_isr`?

**Comment by kaspar030** [comment]
> Perhaps another name? `netdev_event_isr`?

We're bikeshedding again, sorry. `netdev_event_isr` sounds good. Maybe ``netdev_trigger_event_isr`?

**Comment by miri64** [comment]
> We're bikeshedding again, sorry. `netdev_event_isr` sounds good. Maybe `netdev_trigger_event_isr`?

I'd say, @jia200x renames it to one of either and squashes immediately. And then let's go with that.

**Comment by jia200x** [comment]
> `netdev_trigger_event_isr`

I will rename it to that and squash immediately.

**Comment by jia200x** [comment]
done!

**Comment by kaspar030** [comment]
> done!

maybe fix the commit messages

**Comment by jia200x** [comment]
forgot that. Now it should be OK

**Comment by jia200x** [comment]
all green :)

**Comment by jia200x** [comment]
thanks for the review guys!

**Pull request merged**